### DISCUSSION TITLE
ERROR matplotlib.backends.backend_qtagg

### DISCUSSION THREAD

**Discussion by moimaian** [Valdes-Tresanco-MS/gmx_MMPBSA#377] [Q&A] [answered]
Hello. My name is Moisés. I am performing molecular dynamics calculations with GROMACS 2022 and then determining binding free energy with gmx_MMPBSA. I am able to get the result files and access the delta G values. However, a reviewer asked me to graph the binding free energy by frame or time. I realized that in the files we have the values, but I would have to make a script to extract them and mount them in a spreadsheet. The gmx_ana could perhaps provide me with this, but it is showing a bug and closes when trying to display any graph. The error is:
Traceback (most recent call last):
   File "/usr/local/lib/python3.8/dist-packages/GMXMMPBSA/analyzer/customitem.py", line 336, in fn_btn_group
     self.result_table(checked)
   File "/usr/local/lib/python3.8/dist-packages/GMXMMPBSA/analyzer/customitem.py", line 500, in result_table
     from GMXMMPBSA.analyzer.plots import Tables
   File "/usr/local/lib/python3.8/dist-packages/GMXMMPBSA/analyzer/plots.py", line 38, in <module>
     from matplotlib.backends.backend_qtagg import FigureCanvasQTAgg as FigureCanvas, NavigationToolbar2QT
ModuleNotFoundError: No module named 'matplotlib.backends.backend_qtagg'
Aborted (core image saved)

**Comment by Valdes-Tresanco-MS** [comment]
Please, provides more information. What version of gmx_MMPBSA are you using? Show me the output of `conda list` (with the gmxMMPBSA environment activated)

**Reply by moimaian** [reply]
i am using gmx_MMPBSA version 1.5.1
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             4.5                       1_gnu  
amberlite                 16.0                     pypi_0    pypi
ambertools                21.4             py38he7fa2f3_0    conda-forge
amberutils                21.0                     pypi_0    pypi
arpack                    3.7.0                hdefa2d7_2    conda-forge
binutils                  2.35.1               hdd6e379_2    conda-forge
binutils_impl_linux-64    2.35.1               h27ae35d_9  
binutils_linux-64         2.35                h67ddf6f_30    conda-forge
biobb-structure-checking  3.12.1                   pypi_0    pypi
biopython                 1.79                     pypi_0    pypi
boost-cpp                 1.74.0               h312852a_4    conda-forge
brotlipy                  0.7.0           py38h27cfd23_1003  
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.17.1               h7f98852_1    conda-forge
c-compiler                1.2.0                h7f98852_0    conda-forge
ca-certificates           2022.2.1             h06a4308_0  
certifi                   2021.10.8        py38h06a4308_2  
cffi                      1.15.0           py38hd667e15_1  
charset-normalizer        2.0.4              pyhd3eb1b0_0  
compilers                 1.2.0                ha770c72_0    conda-forge
conda                     4.12.0           py38h06a4308_0  
conda-content-trust       0.1.1              pyhd3eb1b0_0  
conda-package-handling    1.7.3            py38h27cfd23_1  
cryptography              36.0.0           py38h9ce1e76_0  
curl                      7.78.0               hea6ffbf_0    conda-forge
cxx-compiler              1.2.0                h4bd325d_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cython                    0.29.24          py38h709712a_0    conda-forge
fftw                      3.3.9           nompi_h74d3f13_101    conda-forge
fortran-compiler          1.2.0                h1990efc_0    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
gcc_impl_linux-64         9.3.0               h70c0ae5_19    conda-forge
gcc_linux-64              9.3.0               hf25ea35_30    conda-forge
gfortran_impl_linux-64    9.3.0               hc4a2995_19    conda-forge
gfortran_linux-64         9.3.0               hdc58fab_30    conda-forge
gxx_impl_linux-64         9.3.0               hd87eabc_19    conda-forge
gxx_linux-64              9.3.0               h3fbe746_30    conda-forge
hdf4                      4.2.15               h10796ff_3    conda-forge
hdf5                      1.10.6          nompi_h6a2412b_1114    conda-forge
icu                       68.1                 h58526e2_0    conda-forge
idna                      3.3                pyhd3eb1b0_0  
jbig                      2.1               h7f98852_2003    conda-forge
jpeg                      9d                   h36c2ea0_0    conda-forge
kernel-headers_linux-64   2.6.32              he073ed8_15    conda-forge
kiwisolver                1.3.1            py38h1fd1430_1    conda-forge
krb5                      1.19.2               hcc1bbae_0    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.35.1               h7274673_9  
lerc                      2.2.1                h9c3ff4c_0    conda-forge
libblas                   3.9.0           11_linux64_openblas    conda-forge
libcblas                  3.9.0           11_linux64_openblas    conda-forge
libcurl                   7.78.0               h2574ce0_0    conda-forge
libdeflate                1.7                  h7f98852_5    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.3                  he6710b0_2  
libgcc-devel_linux-64     9.3.0               h7864c58_19    conda-forge
libgcc-ng                 9.3.0               h5101ec6_17  
libgfortran-ng            9.4.0               h69a702a_14    conda-forge
libgfortran5              9.4.0               h62347ff_14    conda-forge
libgomp                   9.3.0               h5101ec6_17  
liblapack                 3.9.0           11_linux64_openblas    conda-forge
libnetcdf                 4.8.0           nompi_hcd642e3_103    conda-forge
libnghttp2                1.43.0               h812cca2_0    conda-forge
libopenblas               0.3.17          pthreads_h8fe5266_1    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libssh2                   1.9.0                ha56f1ee_6    conda-forge
libstdcxx-devel_linux-64  9.3.0               hb016644_19    conda-forge
libstdcxx-ng              9.3.0               hd4cf53a_17  
libtiff                   4.3.0                hf544144_1    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libwebp-base              1.2.0                h7f98852_2    conda-forge
libxcb                    1.13              h7f98852_1003    conda-forge
libzip                    1.8.0                h4de3113_0    conda-forge
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
matplotlib-base           3.4.2            py38hcc49a3a_0    conda-forge
mmpbsa-py                 16.0                     pypi_0    pypi
ncurses                   6.3                  h7f8727e_2  
netcdf-fortran            4.5.3           nompi_hf3f1587_104    conda-forge
numpy                     1.21.1           py38h9894fe3_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjpeg                  2.4.0                hb52868f_1    conda-forge
openssl                   1.1.1m               h7f8727e_0  
packmol                   20.010               h86c2bf4_0    conda-forge
packmol-memgen            1.2.1rc0                 pypi_0    pypi
pandas                    1.3.1            py38h1abd341_0    conda-forge
parmed                    3.4.1            py38h709712a_0    conda-forge
pdb4amber                 20.1                     pypi_0    pypi
perl                      5.32.1          0_h7f98852_perl5    conda-forge
pillow                    8.3.1            py38h8e6f84c_0    conda-forge
pip                       21.2.4           py38h06a4308_0  
psutil                    5.9.4                    pypi_0    pypi
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
pycosat                   0.6.3            py38h7b6447c_1  
pycparser                 2.21               pyhd3eb1b0_0  
pyopenssl                 21.0.0             pyhd3eb1b0_1  
pyparsing                 3.0.7              pyhd8ed1ab_0    conda-forge
pyqt5                     5.15.6                   pypi_0    pypi
pyqt5-qt5                 5.15.2                   pypi_0    pypi
pyqt5-sip                 12.9.1                   pypi_0    pypi
pysocks                   1.7.1            py38h06a4308_0  
python                    3.8.12               h12debd9_0  
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.8                      2_cp38    conda-forge
pytraj                    2.0.6                    pypi_0    pypi
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
readline                  8.1.2                h7f8727e_1  
requests                  2.27.1             pyhd3eb1b0_0  
ruamel_yaml               0.15.100         py38h27cfd23_0  
sander                    16.0                     pypi_0    pypi
scipy                     1.7.0            py38h7b17777_1    conda-forge
setuptools                58.0.4           py38h06a4308_0  
six                       1.16.0             pyhd3eb1b0_0  
sqlite                    3.37.0               hc218d9a_0  
sysroot_linux-64          2.12                he073ed8_15    conda-forge
tk                        8.6.11               h1ccaba5_0  
tornado                   6.1              py38h497a2fe_1    conda-forge
tqdm                      4.62.3             pyhd3eb1b0_1  
urllib3                   1.26.7             pyhd3eb1b0_0  
wheel                     0.37.1             pyhd3eb1b0_0  
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.0.10               h7f98852_0    conda-forge
xorg-libsm                1.2.3             hd9c2040_1000    conda-forge
xorg-libx11               1.7.2                h7f98852_0    conda-forge
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h7f98852_1    conda-forge
xorg-libxt                1.2.1                h7f98852_2    conda-forge
xorg-xextproto            7.3.0             h7f98852_1002    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.5                h7b6447c_0  
yaml                      0.2.5                h7b6447c_0  
zlib                      1.2.11               h7f8727e_4  
zstd                      1.5.0                ha95c52a_0    conda-forge

**Comment by moimaian** [comment, marked as answer]
I think I got it resolved now. I went to the file /usr/local/lib/python3.8/dist-packages/GMXMMPBSA/analyzer/plots.py and modified line 38, as root , from:
from matplotlib.backends.backend_qtagg import FigureCanvasQTAgg as FigureCanvas, NavigationToolbar2QT
For:
from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas, NavigationToolbar2QT

**Reply by Valdes-Tresanco-MS** [reply]
You can also try to use a newer version by doing a --rewrite-output. This is the advantage of installing gmx_MMPBSA in separate conda environments. You may get some compatibility errors, but they can be solved easily, and I will be happy to help you.

**Comment by moimaian** [comment]
The graphics are all showing now and the program is not closing. I am really happy. Thanks. It was enough to throw the doubt here and the thing flowed. I think I would have to install Qt6 through the miniconda to recognize the library import from the previous description.

**Reply by Valdes-Tresanco-MS** [reply]
This problem we've had for a long time because both gmx_MMPBSA and gmx_MMPBSA_ana are in the same package. This makes cross-version compatibility a headache.
We are solving this problem in the new app. We separated the calculation engine (**xBFReE**) and analysis GUI (**xBFReE-Analyzer**) into two independent packages. With this, we will have a more appropriate development cycle, and we hope to improve in the aspect of compatibility between versions.

**Reply by moimaian** [reply]
I will install the new app. Thanks.
